File: jpa/src/main/java/com/jpa/entity/SalariesEntity.java
package com.jpa.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@Entity
@Table(name = "salaries")
public class SalariesEntity implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "emp_no")
    private Integer empNo;

    @Column(name = "salary")
    private Integer salary;

    @Column(name = "from_date")
    private LocalDate fromDate;

    @Column(name = "to_date")
    private LocalDate toDate;

    //    @Enumerated(EnumType.ORDINAL)
    //    @Column(name = "gender")
    //    private Gender gender;
    @Transient
    @JsonProperty
    private Integer pageSize;

    @Transient
    @JsonProperty
    private Integer currentPage;

}

File: mysql/src/main/kotlin/com/mysql/repository/entity/SalariesEntity.kt
package com.mysql.repository.entity

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable
import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "salaries")
@JsonInclude(JsonInclude.Include.NON_NULL)
data class SalariesEntity (
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "emp_no")
    var empNo: Int,

    @Column(name = "salary")
    var salary: Int? = null,

    @Column(name = "from_date")
    var fromDate: LocalDate? = null,

    @Column(name = "to_date")
    var toDate: LocalDate? = null,

//    @Column(name = "gender")
//    @Enumerated(EnumType.ORDINAL)
//    var gender: Gender? = null,

    //    @Enumerated(EnumType.ORDINAL)
    //    @Column(name = "gender")
    //    private Gender gender;
    @Transient
    @JsonProperty
    var pageSize: Int? = null,

    @Transient
    @JsonProperty
    var currentPage: Int? = null

) : Serializable